### PR TITLE
[Improvement](executor)Refactor Workload group memory GC

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes
Currently when ```Minor GC``` happens, workload group must kill some query to release memory, when Doris upgrade from 2.0 to 2.1 or enable_workload_group, this may cause some query failed which they should success before.
```Minor GC``` is used to reduce the risk of OOM, if origin ```Minor GC``` runs stably, add a workload group gc here is meaningless.

On the contrary, when ```Full GC``` happens, workload group gc first is necessary, because if GC group's overcommit memory can release enough memory, then we can avoid undifferentiated kill query,  this is the purpose of the workload group.
That means if a workload group's query not exceeds group's memory, it can avoid cancel query even ```Full GC``` happens.

### hard limit
```
mysql [hits]>SELECT REGEXP_REPLACE(Referer, '^https?://(?:www\.)?([^/]+)/.*$', '\1') AS k, AVG(length(Referer)) AS l, COUNT(*) AS c, MIN(Referer) FROM hits WHERE Referer <> '' GROUP BY k HAVING COUNT(*) > 100000 ORDER BY l DESC LIMIT 25;
ERROR 1105 (HY000): errCode = 2, detailMessage = (127.0.0.1)[CANCELLED]memory hard limit workload group release overcommit memory, group id:70066, name:test_group, used:699.57 MB, limit:692.70 MB, backend:127.0.0.1. cancel top memory used tracker <Query#Id=736497b7383040d4-bd8ac8c5a1973b34> consumption 699.57 MB. execute again after enough memory, details see be.INFO.

```

### minor GC
```
mysql [hits]>SELECT REGEXP_REPLACE(Referer, '^https?://(?:www\.)?([^/]+)/.*$', '\1') AS k, AVG(length(Referer)) AS l, COUNT(*) AS c, MIN(Referer) FROM hits WHERE Referer <> '' GROUP BY k HAVING COUNT(*) > 100000 ORDER BY l DESC LIMIT 25;
ERROR 1105 (HY000): errCode = 2, detailMessage = (127.0.0.1)[CANCELLED]try cancel memory overcommit query in workload group when MinorGC happens, group id:70066, name:test_group, used:1.10 GB, limit:34.63 MB, backend:127.0.0.1. cancel top memory overcommit tracker <Query#Id=a7abb4ba380c4ea3-b68d3d49bff282ab> consumption 527.02 MB. execute again after enough memory, details see be.INFO.
```

### CODE REVIEW

**Comment by wangbo** [comment]
run buildall

File: be/src/util/mem_info.cpp [hunk 1]
```diff
@@ -147,12 +147,6 @@ bool MemInfo::process_minor_gc() {
         return true;
     }
 
-    RuntimeProfile* tg_profile = profile->create_child("WorkloadGroup", true, true);
-    freed_mem += tg_enable_overcommit_group_gc(_s_process_minor_gc_size - freed_mem, tg_profile);
```
Comment (xinyiZzz) [inline be/src/util/mem_info.cpp:151]: recommended to keep kill query/load in overcommit group.

**Comment by wangbo** [comment]
run buildall

**Review by xinyiZzz — DISMISSED** [review]
LGTM

File: be/src/util/mem_info.h [hunk 2]
```diff
@@ -195,7 +195,7 @@ class MemInfo {
 
     static int64_t tg_not_enable_overcommit_group_gc();
     static int64_t tg_enable_overcommit_group_gc(int64_t request_free_memory,
-                                                 RuntimeProfile* profile);
+                                                 RuntimeProfile* profile, bool is_minor_gc);
```
Comment (xinyiZzz) [inline be/src/util/mem_info.h:198]: only_cancel_overcommit maybe sound better, of course that is ok

**Review by yiguolei — DISMISSED** [review]

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Pull request merged**